sdk/appconfiguration/app-configuration/test/internal/tracingHelpers.spec.ts

```diff
@@ -1,58 +1,184 @@
-    if (newOptions.spanOptions.attributes) {
-      assert.equal("Microsoft.AppConfiguration", newOptions.spanOptions.attributes["az.namespace"]);
-      assert.equal("testAttributeValue", newOptions.spanOptions.attributes["testAttribute"]);
-    }
+    const [status] = setStatusStub!.args[0];
+
+    assert.equal(status.code, CanonicalCode.OK);
+    assert.equal(endStub?.called, true);
   });
 
-  it("getCanonicalCode", () => {
+  it("trace ERROR", async () => {
+    let setStatusStub: sinon.SinonStub<[Status], Span> | undefined;
+    let endStub: sinon.SinonStub | undefined;
+
+    try {
+      await trace(
+        "addConfigurationSetting",
+        {
+          tracingOptions: {}
+        },
+        async (_options: any, _span: Span) => {
+          throw new Error("Purposefully thrown error");
+        },
+        <
+          T extends {
+            tracingOptions?: OperationTracingOptions | undefined;
+          }
+        >(
+          operationName: string,
+          operationOptions: T | undefined
+        ): {
+          span: Span;
+          updatedOptions: T;
+        } => {
+          assert.deepEqual(operationName, "addConfigurationSetting");
+
+          const createdSpan = createSpan(operationName, operationOptions);
+
+          setStatusStub = sinon.stub(createdSpan.span, "setStatus");
+          endStub = sinon.stub(createdSpan.span, "end");
+
+          return createdSpan;
+        }
+      );
+
+      assert.fail("Exception should have been thrown from `trace` since the inner action threw");
+    } catch (err) {
+      assert.equal(err.message, "Purposefully thrown error");
+    }
+
+    assert.ok(setStatusStub, "setStatus should have been called");
     assert.equal(
-      CanonicalCode.PERMISSION_DENIED,
-      Spanner.getCanonicalCode(new RestError("hello", "", 401))
+      setStatusStub?.args[0][0].code,
+      CanonicalCode.INTERNAL,
+      "Any thrown exception causes the span status to be set to an error"
     );
-    assert.equal(
-      CanonicalCode.NOT_FOUND,
-      Spanner.getCanonicalCode(new RestError("hello", "", 404))
+
+    assert.equal(endStub?.called, true);
+  });
+
+  it("tracing is set up for all methods", async () => {
+    const appConfigurationClient = new AppConfigurationClient(
+      "Endpoint=endpoint;Id=id;Secret=secret"
     );
-    assert.equal(CanonicalCode.UNKNOWN, Spanner.getCanonicalCode(new RestError("hello", "", 409)));
-    assert.equal(
-      CanonicalCode.FAILED_PRECONDITION,
-      Spanner.getCanonicalCode(new RestError("hello", "", 412))
+
+    let traceData = {
+      operationName: "",
+      options: undefined as OperationOptions | undefined
+    };
+
+    appConfigurationClient["_trace"] = async (operationName, options, _fn) => {
+      traceData.operationName = operationName;
+      traceData.options = options;
+      return {} as any;
+    };
+
+    const operationOptions: OperationOptions = {
+      abortSignal: ({} as any) as AbortSignalLike,
```

Comment:
Do you even need to set this? `abortSignal` should be optional right?

Reply:
Yeah, you're right. Although in this test it's just cosmetic harmlessness.